From f2226b55287eb1eaa8fd93f23814a2430c8cc6b1 Mon Sep 17 00:00:00 2001
From: honjow <honjow311@gmail.com>
Date: Fri, 13 Mar 2026 02:47:58 +0800
Subject: [PATCH 13/17] config: add mouse to target devices in generic.toml

---
 data/devices/generic.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/devices/generic.toml b/data/devices/generic.toml
index 6457725..6c34358 100644
--- a/data/devices/generic.toml
+++ b/data/devices/generic.toml
@@ -20,4 +20,4 @@ variant = "Generic"
 method = "power_station"
 
 [inputplumber]
-target_devices = ["deck-uhid", "keyboard"]
\ No newline at end of file
+target_devices = ["deck-uhid", "keyboard", "mouse"]
\ No newline at end of file
-- 
2.52.0

